### DISCUSSION TITLE
Snap installation an serial Interface permissions

### DISCUSSION THREAD

**Discussion by Hartislap** [wmbusmeters/wmbusmeters#931] [Q&A]
Hi,

i have a working installation of wmbusmeters with my serial mbus interface and wanted to switch to an new raspi. For the installation I switched from the compiled version to the snap installation as it was suggested in the wiki article.

The problem is I cannot read the mbus using the snap installed version:
(wmbusmeters) version: 1.12.0-15-g3177c44
(config) store meter files in: "/var/snap/wmbusmeters/common/logs/meter_readings"
(config) using device: /dev/serial0:mbus:2400 
(config) number of meters: 0
(lookup) with file/hex "/dev/serial0" mbus
(lookup) driver: mbus
(main) opening /dev/serial0:mbus:2400
Started config mbus on /dev/serial0
(mbus) on /dev/serial0
(serial) EVENT thread interrupted
(serialtty) could not open /dev/serial0 with 2400 baud N81
(main) regular reset of mbus  on /dev/serial0 will happen every 82800 seconds
(wmbus) no alarm (expected activity) for mbus
(serial) registered regular callback HOT_PLUG_DETECTOR(0) every 2 seconds
No meters configured. Printing id:s of all telegrams heard!
(serial) waiting for stop
(serial) EVENT thread interrupted
(serial) TIMER thread interrupted

On the old installation:
(wmbusmeters) version: 1.9.0-78-g86b497d
(config) store meter files in: "/var/lib/wmbusmeters/meter_readings"
(config) using device: /dev/serial0:mbus:2400 
(config) number of meters: 2
(meter) created UH50MBus ultraheat 68386406 not-encrypted
(meter) created ITRONMBus auto 21012202 not-encrypted
(lookup) with file/hex "/dev/serial0" mbus
(lookup) driver: mbus
(main) opening /dev/serial0:mbus:2400
[2023-03-28_09:02:54] Started config mbus on /dev/serial0
(mbus) on /dev/serial0

My guess: the snap installation is not allowed to access the serial port? I found only the explanation for USB-Devices using "sudo snap connect wmbusmeters:raw-usb core:raw-usb", but maybe I need to connect the serial port instead?
When I have a look at the snap yaml there are only those 4 plugs:
      - network
      - network-bind
      - raw-usb
      - system-observe
Is there a way to get this installation working with a serial interface or is the only way to compile it on the new raspi?

Cheers
Philipp

**Comment by BIBOLV** [comment]
Hi,
Added serial plug to config, once images will be built you can test. Before testing do not forget to connect plug to snap.
Since mbus support is quite new, it has not been tested and implemented everywhere. BTW, if you are up for it - you could also test docker container. Does it work with mbus or it also needs some config adjustment.
